GlimpseImputationPipeline/Glimpse2MergeBatches.wdl

```diff
@@ -146,6 +356,138 @@ task CountSamples { # really?
+if len(num_samples) != len(input_filenames):
+    raise RuntimeError('The number of input annotations does not match the number of input number of samples.')
+num_batches = len(input_filenames)
+
+def calculate_af(row):
+    return sum([row[f'AF_{i}'] * num_samples[i] for i in range(num_batches)]) / sum(num_samples)
+def calculate_info(row):
+    aggregated_af = row['AF']
+    return 1 if aggregated_af == 0 or aggregated_af == 1 else \
+                     1 - \
+                    (sum([(1 - row[f'INFO_{i}']) * 2 * num_samples[i] * row[f'AF_{i}'] * (1 - row[f'AF_{i}']) for i in range(num_batches)])) / \
+                    (2 * sum(num_samples) * aggregated_af * (1 - aggregated_af))
+
+annotation_dfs = [pd.read_csv(input_filename, sep='\t').rename(columns={'AF': f'AF_{i}', 'INFO': f'INFO_{i}'}) for i, input_filename in enumerate(input_filenames)]
+annotations_merged = functools.reduce(lambda left, right: pd.merge(left, right, on=['CHROM', 'POS', 'REF', 'ALT'], how='inner', validate='one_to_one'), annotation_dfs)
+annotations_merged['AF'] = annotations_merged.apply(lambda row: calculate_af(row), axis=1)
+annotations_merged['INFO'] = annotations_merged.apply(lambda row: calculate_info(row), axis=1)
+annotations_merged.to_csv('aggregated_annotations.tsv', sep='\t', columns=['CHROM', 'POS', 'REF', 'ALT', 'AF', 'INFO'], header=False, index=False)
+
+EOF
+        python3 script.py
+
+        bgzip aggregated_annotations.tsv
+        tabix -s1 -b2 -e2 aggregated_annotations.tsv.gz
+
+        bcftools annotate -a aggregated_annotations.tsv.gz -c CHROM,POS,REF,ALT,AF,INFO -O z -o ~{output_basename}.vcf.gz ~{merged_vcf}
+    >>>
+
+    runtime {
+        docker: docker_merge
+        disks: "local-disk " + disk_size_gb + " HDD"
+        memory: mem_gb + " GiB"
+        cpu: cpu
+        preemptible: preemptible
+    }
+
+    output {
+        File merged_imputed_vcf = "~{output_basename}.vcf.gz"
+        File aggregated_annotations = "aggregated_annotations.tsv.gz"
+    }
+}   
+
+task MergeQCMetrics {
+    input {
+        Array[File] qc_metrics
+        String docker_merge
+        String output_basename
+
+        Int disk_size_gb = ceil(2.2 * size(qc_metrics, "GiB") + 50)
+        Int mem_gb = 4
+        Int cpu = 2
+        Int preemptible = 1
+    }
+
+    command <<<
+        set -xeuo pipefail
+        
+
+        cat <<EOF > script.py
+    import pandas as pd
+    qc_metrics = ['~{sep="', '" select_first([qc_metrics, []])}']
+    merged_qc_metrics = pd.concat([pd.read_csv(qc_metric, sep='\t') for qc_metric in qc_metrics])
+    merged_qc_metrics.to_csv('~{output_basename}.qc_metrics.tsv', sep='\t')
+    EOF
+    >>>
+
+    output {
+        File merged_qc_metrics = "~{output_basename}.qc_metrics.tsv"
+    }
+
+    runtime {
+        docker: docker_merge
+        disks: "local-disk " + disk_size_gb + " HDD"
+        memory: mem_gb + " GiB"
+        cpu: cpu
+        preemptible: preemptible
+    }
+}
+
+
 task MergeAndRecomputeAndAnnotate {
```

Comment:
Can get rid of this entire task. The only other place this is referenced is in a comment which you could update as well then, depending on which option you choose about QC metrics being required or optional